plugins/main/tap/distro/rhel/rhel.go

```diff
@@ -0,0 +1,87 @@
+package rhel
+
+import (
+	"fmt"
+	"github.com/containernetworking/plugins/plugins/main/tap/distro"
+	"github.com/opencontainers/selinux/go-selinux"
+	"os/exec"
+	"strconv"
+	"strings"
+	"syscall"
+)
+
+var Rhel distro.Distro = CreateLink{}
+
+type CreateLink struct{}
+
+func (l CreateLink) CreateLink(tmpName string, mtu int, nsFd int, nsPath string, multique bool, mac string, owner int, group int) error {
+
+	err := setContainerSeBool()
+	if err != nil {
+		return err
+	}
+	err = createSelinuxTap(tmpName, mtu, nsFd, nsPath, multique, mac, owner, group)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func setContainerSeBool() error {
+	output, err := exec.Command("cat", "/sys/fs/selinux/booleans/container_use_devices").CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("failed to run getsebool command %s: %v", string(output), err)
+	}
+	if strings.Contains(string(output), "off") {
+		output, err := exec.Command("setsebool", "-P", "container_use_devices", "true").CombinedOutput()
+		if err != nil {
+			return fmt.Errorf("failed to run setsebool command %s: %v", string(output), err)
+		}
+	}
+	return nil
+}
+
+// Due to issues with the vishvananda/netlink library (fix pending) this method is using the ip tool to set up
+// the tap device.
+func createSelinuxTap(tmpName string, mtu int, nsFd int, nsPath string, multiqueue bool, mac string, owner int, group int) error {
+	if err := selinux.SetExecLabel("system_u:system_r:container_t:s0"); err != nil {
```

Comment:
@mmirecki couldn't this just be a config file option, such that each user could customize it for their specific distro if it differs between distros/policies?

Reply:
Done

Reply:
@mmirecki does this play along w/ MCS (Multi-Category Security) ? IIUC, if the exec labels on a pod feature MCS, they won't be able to open a generic `container_t` label.

Reply:
@maiqueb I'm not an expert on MCS, but note that this code is not executed in a pod. The CNI plugin only operates on the pod's network namespace.

Reply:
yes, but it ends up being the label for that particular container; consider for instance KubeVirt launcher pods (these encapsulate the VM): these will have the following label: `system_u:object_r:container_t:s0:c30,c157` . 

This would cause the processes running in this container to fail accessing resources with the `container_t` label.

AFAIU, this is in place so one container cannot access resources of **another** container.

If we'd just used the `container_t` label, any container (running without MCS, thus running w/ the `container_t` label) would be able to consume the taps of any other container - something which I think we don't want to allow.

Reply:
The custom label would allow the user to either use the label and resolve this in the plugin, or resolve it outside the plugin.
I'm not an expert on the MCS, but from what I read it looks like it's not widely used. I think it should be enough to support the targeted mode for selinux (or have a custom label for mcs?)

Reply:
If possible, we'd love to use this CNI from KubeVirt.

Thing is, we use MCS...

I'm just saying that this approach (user provided label) will quite probably be impossible to use for MCS-aware workloads. @jean-edouard can you keep me honest here ?

Having said that, it could be OK to move forward with this proposal now, and investigate how to account for MCS later on - I think we may be able to pass the MCS part of the string via CNI_ARGS or runtime config.

Reply:
@maiqueb Could you please suggest a way to also satisfy the MCS way?

Reply:
After speaking w/ @jean-edouard I was corrected on my understanding of MCS:
- a container using an MCS-aware label - i.e. `container_t:s0:c1,c2` **can** access a `container_t:s0` resource.

The only issue here would be the security risk of **any** container being able to consume taps created by another container - not optimal.

I think we can move forward with this implementation from my perspective.

We can tackle the MCS functionality as a follow-up - I think we can pass the label via runtime config / CNI args to the plugin.